container/container_unix.go

```diff
@@ -410,28 +410,50 @@ func copyOwnership(source, destination string) error {
 	return os.Chmod(destination, os.FileMode(stat.Mode()))
 }
 
+// validateTmpfsMountDestination validates the destination of tmpfs mount.
+// Currently, we have only two obvious rule for validation:
+//  - path must be absolute
+//  - path must not be "/"
+// We should add more rules carefully (#30166)
+func validateTmpfsMountDestination(dest string) error {
+	cleanedDest := filepath.Clean(dest)
+	if !filepath.IsAbs(cleanedDest) {
+		return fmt.Errorf("mount path for tmpfs must be absolute: %q", dest)
+	}
+	prohibited := []string{"/"}
+	for _, s := range prohibited {
+		if s == cleanedDest {
+			return fmt.Errorf("unexpected mount path for tmpfs: %q", dest)
+		}
+	}
+	return nil
+}
+
 // TmpfsMounts returns the list of tmpfs mounts
 func (container *Container) TmpfsMounts() ([]Mount, error) {
-	var mounts []Mount
+	m := make(map[string]string)
 	for dest, data := range container.HostConfig.Tmpfs {
-		mounts = append(mounts, Mount{
-			Source:      "tmpfs",
-			Destination: dest,
-			Data:        data,
-		})
+		m[dest] = data
 	}
 	for dest, mnt := range container.MountPoints {
 		if mnt.Type == mounttypes.TypeTmpfs {
 			data, err := volume.ConvertTmpfsOptions(mnt.Spec.TmpfsOptions, mnt.Spec.ReadOnly)
 			if err != nil {
 				return nil, err
 			}
-			mounts = append(mounts, Mount{
-				Source:      "tmpfs",
-				Destination: dest,
-				Data:        data,
-			})
+			m[dest] = data
+		}
+	}
+	var mounts []Mount
+	for dest, data := range m {
+		if err := validateTmpfsMountDestination(dest); err != nil {
```

Comment:
Given that we're already looping through `container.HostConfig.Tmpfs` and `container.MountPoints`; perhaps put this validation step inside those loops?